platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/transition/execution/auto/WorkflowAutoTransitionExecutor.java

```diff
@@ -27,44 +27,61 @@ public WorkflowAutoTransitionExecutor(AGWorkflowItem item, Collection<AGWorkflow
 	public void evaluateAndExecute() {
 
 		try (DbTransaction transaction = new DbTransaction()) {
-			checkSourceNodeOfTransitions();
-			List<AGWorkflowTransition> validTransitions = transitions//
-				.stream()
-				.filter(it -> validateNodePreconditions(item, it))
-				.collect(Collectors.toList());
-			if (validTransitions.size() > 1) {
-				List<IDisplayString> transitionTitles = validTransitions//
+			item.reloadForUpdate();
+			if (checkSourceNodeOfTransitions()) {
+				List<AGWorkflowTransition> validTransitions = transitions//
 					.stream()
-					.map(AGWorkflowTransition::toDisplayWithoutId)
+					.filter(it -> validateNodePreconditions(item, it))
 					.collect(Collectors.toList());
-				throw new SofticarUserException(//
-					WorkflowI18n.WORKFLOW_ITEM_ARG1_HAS_MORE_THAN_ONE_EXECUTABLE_AUTO_TRANSITION_ARG2
-						.toDisplay(item.toDisplayWithoutId(), Imploder.implode(transitionTitles, "\n")));
-			} else if (validTransitions.size() == 1) {
-				var transition = validTransitions.get(0);
-				var targetNode = transition.getTargetNode();
-				Log
-					.finfo(//
-						"Executing transition '%s' into target node '%s' for item '%s'.",
-						transition.toDisplayWithoutId(),
-						targetNode.toDisplayWithoutId(),
-						item.toDisplay());
-				transition.executeSideEffect(item);
-				new WorkflowTaskManager(item).setNextNodeAndGenerateTasks(targetNode);
-				new AGWorkflowAutoTransitionExecution().setWorkflowItem(item).setWorkflowTransition(transition).save();
-			} else {
-				Log.fverbose("No executable auto transition found.");
+				evaluateAndExecute(validTransitions);
 			}
 			transaction.commit();
 		}
 	}
 
-	private void checkSourceNodeOfTransitions() {
+	private void evaluateAndExecute(List<AGWorkflowTransition> validTransitions) {
```

Comment:
To improve encapsulation and legibility, we could as well load the transitions inside this method, instead of passing them as a parameter. Like this:

```
	public void evaluateAndExecute() {

		try (DbTransaction transaction = new DbTransaction()) {
			boolean reloaded = item.reloadForUpdate();
			if (!reloaded) {
				Log.fwarning("Workflow item '%s' could not be reloaded.", item.getId());
			} else if (allTransitionsHaveExpectedSourceNode()) {
				loadAndExecuteTransitions();
			}
			transaction.commit();
		}
	}

	private void loadAndExecuteTransitions() {

		List<AGWorkflowTransition> validTransitions = transitions//
			.stream()
			.filter(it -> validateNodePreconditions(item, it))
			.collect(Collectors.toList());

		if (validTransitions.size() > 1) {
		// ...
```
Note that I renamed the now-parameterless method to `loadAndExecuteTransitions`, in order to avoid a name clash.

Reply:
done